### PR TITLE
Sborse/nav2 behavior

### PR DESCRIPTION
**Pull request by borsesaurabh2022**
Pull request for recovery behavior:

### Change Content: (New package: recovery_behavior)
1. recover_pose_action_client:  Action client for bt_navigator to request for recover pose action 
2. recover_pose: behavior server, action server to execute the recovery action requested by the bt_navigator 
    - Collision check using local and global cost-map before executing the recovery 
    - Dynamic collision check while executing the recovery behavior 
2. backup:  behavior server, action server to execute the backup recovery action requested by the bt_navigator 
    - Collision check using local and global cost-map before executing the recovery 
    - Dynamic collision check while executing the recovery behavior

### Change Content: (New package: recovery_msgs)
created action msgs for a new recovery behavior - recover_pose
 
### Change Content: (package: robotino_navigation)
1. Custom behavior tree for use case
2. Added sca_irsensor, another source of sensor readings for obstacle layer of local cos-tamp and fixed object detection/ ray-trace parameter for object detection and clearing

### CODE REVIEW

File: recovery_behavior/plugins/action/recover_pose_action.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
```
Comment (TarikViehmann) [inline recovery_behavior/plugins/action/recover_pose_action.cpp:1]: removce long license header, keep line 1

File: recovery_behavior/include/recovery_behavior/plugins/action/recover_pose_action.hpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
+
+// Copyright (c) 2018 Intel Corporation
```
Comment (TarikViehmann) [inline recovery_behavior/include/recovery_behavior/plugins/action/recover_pose_action.hpp:3]: remove long license header, keep line 1

also, why is there Intel Corporation?!!?

File: recovery_behavior/include/recovery_behavior/back_up.hpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
+
```
Comment (TarikViehmann) [inline recovery_behavior/include/recovery_behavior/back_up.hpp:2]: remove long license header, keep line 1

**Review by TarikViehmann — CHANGES_REQUESTED** [review]

File: recovery_behavior/include/recovery_behavior/back_up.hpp (hunk 3 above)
Comment (borsesaurabh2022) [reply recovery_behavior/include/recovery_behavior/back_up.hpp:2]: fixed, pls refer to latest commit [dce5e3e](https://github.com/carologistics/robotino_navigation/pull/14/commits/dce5e3ee9f1a73e0995f00660a3e138e691fc48

File: recovery_behavior/include/recovery_behavior/plugins/action/recover_pose_action.hpp (hunk 2 above)
Comment (borsesaurabh2022) [reply recovery_behavior/include/recovery_behavior/plugins/action/recover_pose_action.hpp]: fixed, pls refer to latest commit - dce5e3e

File: recovery_behavior/include/recovery_behavior/recover_pose.hpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
```
[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply recovery_behavior/include/recovery_behavior/recover_pose.hpp:1]: fixed, pls refer to latest commit - dce5e3e

File: recovery_behavior/plugins/action/recover_pose_action.cpp (hunk 1 above)
Comment (borsesaurabh2022) [reply recovery_behavior/plugins/action/recover_pose_action.cpp:1]: fixed, pls refer to commit - dce5e3e

File: recovery_behavior/src/back_up.cpp [hunk 5]
```diff
@@ -0,0 +1,227 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
+
+//   created on: 2024-06-02
```
[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply recovery_behavior/src/back_up.cpp]: fixed, pls refer to latest commit - dce5e3e

File: recovery_behavior/src/recover_pose.cpp [hunk 6]
```diff
@@ -0,0 +1,279 @@
+// Licensed under MIT. See LICENSE file. Copyright Carologistics.
+
+// created on: 2024-06-02
+// MIT License
+//
+// Copyright (c) 2023 Saurabh Borse
```
[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply recovery_behavior/src/recover_pose.cpp]: fixed, pls refer to latest commit - dce5e3e

File: recovery_msgs/LICENSE [hunk 7]
```diff
@@ -0,0 +1,202 @@
+
```
[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply recovery_msgs/LICENSE]: fixed, pls refer to the latest commit - dce5e3e

File: robotino_navigation/config/robotinobase1_mps_map_gen.yaml [hunk 8]
```diff
@@ -25,8 +25,10 @@ local_costmap:
       global_frame: robotinobase1/odom
       robot_base_frame: robotinobase1/base_link
       obstacle_layer:
-        scan:
+        scan_lidar:
```
[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply robotino_navigation/config/robotinobase1_mps_map_gen.yaml:28]: Done, updated the param for rb4, refer commit- [e9bbc1b]

File: recovery_behavior/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+find_package(geometry_msgs REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(nav2_costmap_2d REQUIRED)
+find_package(nav2_core REQUIRED)
+find_package(nav2_behaviors REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
+find_package(action_msgs REQUIRED)
+find_package(CURL REQUIRED)
```
[the remark this answers is not in the document]
Comment (TarikViehmann) [reply recovery_behavior/CMakeLists.txt:32]: This one still needs to be adressed

[the remark this answers is not in the document]
Comment (borsesaurabh2022) [reply recovery_behavior/CMakeLists.txt]: done, updated Cmakelist and packagexml, refer commit [49bc7c9](https://github.com/carologistics/robotino_navigation/pull/14/commits/49bc7c97fd5f8b364359dcd2989f2838184040ee)

**Review by TarikViehmann — APPROVED** [review]
Thank you, good job!

**Pull request merged**